xdem/coreg.py

```diff
@@ -468,6 +469,20 @@ def fit(self, reference_dem: Union[np.ndarray, np.ma.masked_array],
         if weights is not None:
             raise NotImplementedError("Weights have not yet been implemented")
 
+        # If both DEMs are Rasters, validate that 'dem_to_be_aligned' is in the right grid. Then extract its data.
+        if isinstance(dem_to_be_aligned, gu.Raster) and isinstance(reference_dem, gu.Raster):
+            dem_to_be_aligned = dem_to_be_aligned.reproject(reference_dem, silent=True).data
+        # If 'reference_dem' is not a Raster, just extract the data (shape validation comes later)
+        elif isinstance(dem_to_be_aligned, gu.Raster):
+            dem_to_be_aligned = dem_to_be_aligned.data
+
+        # If 'reference_dem' is a Raster, use its transform (if not given) and then extract its data.
+        if isinstance(reference_dem, gu.Raster):
+
+            if transform is None:
```

Comment:
I wonder if maybe we should raise a RunTimeWarning if a `RasterType` is provided but another `transform` is given and used? 
That seems quite dangerous to me, and the user should be informed! 
And, in practice, it won't happen so often that the Warning will become annoying. It's really rare to provide a different transform than that of your Raster

Reply:
Added in cc0cdb3!

I accidentally included it with a merge... Jaja, I blame the 3 AM birds!